capstone-app/src/app/modules/maps/map/map.component.ts

```diff
@@ -27,47 +42,122 @@ export class MapComponent {
     setTimeout(() => {
       this.locationSearch();
     });
+    this.nearSearch = new google.maps.places.PlacesService(this.mapComponent._googleMap);
   }
 
   locationSearch() {
     const input = this.searchBar.nativeElement;
+    this.cityLocation = this.searchBar.nativeElement.value;
     const autoComplete = new google.maps.places.Autocomplete(input,
-      {fields: ['geometry', 'name'], types: ['(cities)']});
+      {fields: ['geometry', 'name', 'formatted_address'], types: ['(cities)']});
     autoComplete.addListener('place_changed', () => {
       this.location = autoComplete.getPlace().geometry.location;
+      this.cityLocation = autoComplete.getPlace().formatted_address;
+      this.placesRequestFunc(this.location);
     });
   }
 
+  deleteAllMarks() {
+    this.searchMarkers = [];
+  }
+
+  placesRequestFunc(location: google.maps.LatLng) {
+    this.placesRequest.location = this.location;
+    this.nearSearch.nearbySearch(this.placesRequest, results => {
+      // resetting data
+      this.deleteAllMarks();
```

Comment:
This function is only called once and only has one line. Do we need it to be a function?

Reply:
this function is call every time we use the input box which then triggers placeRequestFunc(). So I end up changing this method which end up making shorter but before it was longer therefore, i exported it to another method.